pkg/rbd/controllerserver.go

```diff
@@ -411,8 +405,17 @@ func (cs *ControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 	idLk := snapshotNameLocker.Lock(req.GetName())
 	defer snapshotNameLocker.Unlock(idLk, req.GetName())
 
-	// Need to check for already existing snapshot name, and if found
-	// check for the requested source volume id and already allocated source volume id
+	// updating the rbdImage name to point to temparory cloned image name, this
+	// is to check if the snapshot is already present or not
+	img, err := getCloneImageName(rbdSnap, cr)
+	if err != nil {
+		if _, ok := err.(ErrImageNotFound); !ok {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+	} else {
+		rbdSnap.RbdImageName = img
+	}
+
 	found, err := checkSnapExists(rbdSnap, cr)
```

Comment:
If the "snapshot" RBD image wasn't found above, why do we need to check if the snapshot exists in the metadata? It just seems odd how the `rbdSnap.RbdImageName` field is being manipulated. Maybe if it wasn't set on line 400 and reset on line 443 it would be clearer. It seems like it should always only be set to the snapshot image name.

Reply:
at line 400 am setting the rbd image name to parent Volume from which the snapshot has to be created.

as the snapshot will be created from a cloned image, not the parent image (i need to set the snaps rbd name to temp cloned image and check if the snapshot is already present )

so am checking if the temp image at line 410, if its present am updating the snapshot parent image name as temp cloned image name at line 416 and checking if snapshot already created

as snapshot parent image name is updated to temp image at line 416. at line 443 resetting back to the parent vol.